gcf/lib/path.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package lib
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"path/filepath"
+	"strings"
+)
+
+// DataFiles represent a logical unit of files for a single dataset.
+type DataFiles struct {
+	TMCFPath string
+	CSVPaths []string
+}
+
+// ImportGroupFiles represent the collection of paths for a single import group.
+type ImportGroupFiles struct {
+	DataSource2DatasetNames map[string][]string
+	DatasetName2DataFiles   map[string]DataFiles
+	ImportGroupName         string
+}
+
+// CollectImportFiles takes a list of gcs file paths and construct ImportGroupFiles.
+func CollectImportFiles(paths []string) (*ImportGroupFiles, error) {
+
+	dataSource2DatasetNames := map[string][]string{}
+	datasetName2DataFiles := map[string]DataFiles{}
+
+	if len(paths) == 0 {
+		return nil, errors.New("No path found in import group files")
+	}
+	// IMPORTANT NOTE: importGroupName has a character limit of 21
+	importGroupName := strings.Split(paths[0], "/")[0]
+	if len(importGroupName) > 20 {
+		importGroupName = importGroupName[:20]
+	}
+
+	for _, path := range paths {
+		// Path is expected to be like the following:
+		// 	"demo/data/source2/solar/output.csv"
```

Comment:
line54 missing a subfolder?

Reply:
That is intentional, <dataset>/a.tmcf and <dataset>/subfolder/a.tmcf are both valid.

Reply:
oh, actually line 52 "subfolder" is not supported?

Reply:
added a constraint.